src/NServiceBus.Core/Recoverability/Recoverability.cs

```diff
@@ -58,98 +52,79 @@ protected internal override void Setup(FeatureConfigurationContext context)
                         {Headers.HostDisplayName, hostInfo.DisplayName}
                     };
 
-                    return new MoveToErrorsExecutor(b.Build<IDispatchMessages>(), errorQueue, staticFaultMetadata);
+                    var headerCustomizations = context.Settings.Get<Action<Dictionary<string, string>>>(FaultHeaderCustomization);
+
+                    return new MoveToErrorsExecutor(b.Build<IDispatchMessages>(), errorQueue, staticFaultMetadata, headerCustomizations);
                 };
 
-                var delayedRetriesEnabled = IsDelayedRetriesEnabled(context.Settings);
+                var transactionsOn = context.Settings.GetRequiredTransactionModeForReceives() != TransportTransactionMode.None;
+                var delayedRetryConfig = GetDelayedRetryConfig(context.Settings, transactionsOn);
                 Func<string, DelayedRetryExecutor> delayedRetryExecutorFactory = localAddress =>
                 {
-                    if (!delayedRetriesEnabled || transportTransactionMode == TransportTransactionMode.None)
+                    if (!transactionsOn)
                     {
                         return null;
                     }
                     return new DelayedRetryExecutor(
```

Comment:
We should not be creating `DelayedRetriesExecutor` when neither native delays nor TM is available.